R/zchunk_L244.building_USA.R

```diff
@@ -0,0 +1,910 @@
+      # Calculate base.service = calibrated.value(energy) * efficiency
+      mutate(base.service = round(calibrated.value * efficiency, energy.DIGITS_CALOUTPUT)) %>%
+      # Aggregate base service by service (supplysector)
+      group_by(region, supplysector, year) %>%
+      summarise(base.service = sum(base.service)) %>%
+      ungroup() %>%
+      # Add gcam.consumer (sector)
+      left_join_error_no_match(calibrated_techs_bld_usa_consumer, by = "supplysector") %>%
+      # Add nodeInput and building.node.input
+      left_join_error_no_match(A44.gcam_consumer, by = "gcam.consumer")
+
+    # Separate thermal and generic services into separate tables with different ID strings
+    L244.GenericBaseService_gcamusa <- L244.base_service %>%
+      filter(supplysector %in% generic_services) %>%
+      rename(building.service.input = supplysector) %>%
+      select(LEVEL2_DATA_NAMES[["GenericBaseService"]])
+
+    L244.ThermalBaseService_gcamusa <- L244.base_service %>%
+      filter(supplysector %in% thermal_services) %>%
+      rename(thermal.building.service.input = supplysector) %>%
+      select(LEVEL2_DATA_NAMES[["ThermalBaseService"]])
+
+    # L244.GenericServiceSatiation_gcamusa: Satiation levels assumed for non-thermal building services
+    # Just multiply the base-service by an exogenous multiplier
+    L244.GenericServiceSatiation_gcamusa <- L244.GenericBaseService_gcamusa %>%
+      filter(year == max(BASE_YEARS))
+
+    # When adding floorspace, we should take floorspace from max(BASE_YEARS) as well
+    # Instead we take the first value, which ends up being floorspace from min(BASE_YEARS)
+    if(OLD_DATA_SYSTEM_BEHAVIOR){
+      L244.GenericServiceSatiation_gcamusa <- L244.GenericServiceSatiation_gcamusa %>%
+        # Add floorspace
+        left_join_keep_first_only(L244.Floorspace_gcamusa, by = c(LEVEL2_DATA_NAMES[["BldNodes"]]))
+    } else {
+      L244.GenericServiceSatiation_gcamusa <- L244.GenericServiceSatiation_gcamusa %>%
+      # Add floorspace
+      left_join_error_no_match(L244.Floorspace_gcamusa, by = c(LEVEL2_DATA_NAMES[["BldNodes"]], "year"))
+    }
+    L244.GenericServiceSatiation_gcamusa <- L244.GenericServiceSatiation_gcamusa %>%
+      # Add multiplier
+      left_join_error_no_match(A44.demand_satiation_mult, by = c("building.service.input" = "supplysector")) %>%
+      # Satiation level = service per floorspace * multiplier
+      mutate(satiation.level = round(base.service / base.building.size * multiplier, energy.DIGITS_COEFFICIENT)) %>%
+      select(LEVEL2_DATA_NAMES[["GenericServiceSatiation"]])
+
+
+    # L244.ThermalServiceSatiation: Satiation levels assumed for thermal building services
+    L244.ThermalServiceSatiation_gcamusa <- L244.ThermalBaseService_gcamusa %>%
+      filter(year == max(BASE_YEARS))
+
+    # When adding floorspace, we should take floorspace from max(BASE_YEARS) as well
```

Comment:
I don't think I follow this. Can you explain a bit here?

Reply:
We filter `L244.ThermalBaseService_gcamusa` to `max(BASE_YEARS)`, but then when we join in floorspace, we don't specify a year, so it just takes the first year in floorspace. Basically we join 2010 service values with 1975 floorspace values. If we include `"year"`  in the matching, then both values are from 2010

Reply:
Please incorporate that into code comment as practical/useful. Thanks.